gadget-container/runchooks/runchooks.c

```diff
@@ -0,0 +1,64 @@
+#define _GNU_SOURCE
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+
+static void __myinit(void) __attribute__((constructor));
+static void __myinit(void)
+{
+	// ignore non-root programs
+	if (geteuid() != 0)
+		return;
+
+	// ignore non-runc programs
+	char buf[1024] = {0,};
+	int ret;
+	ret = readlink("/proc/self/exe", buf, sizeof(buf));
+	if (ret <= 0 || ret == sizeof(buf))
+		return;
+	char *b = basename(buf);
+	if (strcmp(b, "runc") != 0) {
+		return;
+	}
+
+	// find the bundle directory in the command line
+	FILE *cmdline = fopen("/proc/self/cmdline", "r");
+	char *arg = 0;
+	size_t size = 0;
+	int create_found = 0;
+	int next = 0;
+	char *bundledir = NULL;
+	while(getdelim(&arg, &size, 0, cmdline) != -1) {
+		if (strcmp("create", arg) == 0) {
+			create_found = 1;
+			continue;
+		}
+		if (create_found && strcmp("--bundle", arg) == 0) {
+			next = 1;
+			continue;
+		}
+		if (next) {
+			bundledir = strdup(arg);
+			break;
+		}
+	}
+	free(arg);
+	fclose(cmdline);
+
+	// ignore runc programs for other verbs than 'create'
+	if (!create_found)
+		return;
+
+	// ignore runc programs without bundle directory
+	if (bundledir == NULL)
+		return;
+
+	char cmd[1024] = {0,};
+
+	ret = snprintf(cmd, sizeof(cmd), "cd %s && cp config.json config.json.orig && cat config.json.orig | jq -r \". * `cat /opt/runchooks/add-hooks.jq`\" > config.json", bundledir);
```

Comment:
For the reference https://github.com/kinvolk/inspektor-gadget/pull/41#discussion_r417412917

@alban what do you think about using a C json library to do that work?

Reply:
Right, in that case we could create an independent executable that uses the json library to do the work and then invoke from here with `system()`. I have never used any C json library, from the ones linked in https://www.json.org/json-en.html I saw there are many that are composed by a single source and header file, so we could take and compile the code directly. 

Given that it's a hack, maybe the only thing we could do now is to ignore any error if jq is not present on the system. Or do you think it's a good idea to make this mechanism more robust and get rid of the jq dependency?